js/modules/k6/ws/ws.go

```diff
@@ -286,8 +286,14 @@ func (mi *WS) Connect(url string, args ...goja.Value) (*WSHTTPResponse, error) {
 	if connErr != nil {
 		// Pass the error to the user script before exiting immediately
 		socket.handleEvent("error", rt.ToValue(connErr))
-
-		return nil, connErr
+		if state.Options.Throw.Bool {
+			return nil, connErr
+		} else {
+			if state.Logger != nil {
+				state.Logger.WithField("error", connErr).Warn("Ws Connection Failed")
+			}
+			return &WSHTTPResponse{}, nil
```

Comment:
Part of the issue is to also return a response containing the error so that the javascript code can do something based on that.

This kinds of threads on https://github.com/grafana/k6/issues/2024 as well and adding an `error_code` and `error` will require (as mentioned in that issue) for the error code to be moved somewhere and exported. This likely is a bit too much work so I am okay with not being part of this PR. 

But we definitely want the response to at least have an `error` field.